src/Form/Form.js

```diff
@@ -3,27 +3,54 @@ import classes from "./Form.module.css";
 import * as firebase from "../lib/firebase";
 
 const Form = props => {
+  const { setdbItems, dbItems, token, onEnterToken } = props;
   const [name, setName] = useState("");
   const [frequency, setFrequency] = useState(0);
   const [date, setDate] = useState("");
 
   const handleSubmit = e => {
     e.preventDefault();
     let db = firebase.fb.firestore();
-    let data = {
-      name,
-      frequency,
-      date
-    };
-    db.collection(props.token)
-      .add(data)
-      .then(function() {
-        alert("Item added!");
-      })
-      .catch(function(error) {
-        alert("Something went wrong!");
-        console.error("Error writing document: ", error);
-      });
+    let newName = name
+      .toUpperCase()
+      .replace(/[^\w\s]|_/g, "")
+      .replace(/\s+/g, " ");
+    const existingItem = dbItems.find(element => {
+      let newElement = element.name
+        .toUpperCase()
+        .replace(/[^\w\s]|_/g, "")
+        .replace(/\s+/g, " ");
+      return newElement === newName;
+    });
+
+    if (!existingItem) {
+      let data = {
+        name,
+        frequency,
+        date
+      };
+      db.collection(token)
+        .add(data)
+        .then(function() {
+          alert("Item added!");
+        })
+        .catch(function(error) {
+          alert("Something went wrong!");
+          console.error("Error writing document: ", error);
+        })
+        .then(
+          db
```

Comment:
I think this logic could be drastically reduced and simplified if we used the `react-firestore` component to use websockets to stay constantly connected to the database, thus always having the most recent data. We wouldn't need to fetch the collection once more here, because the library would have already re-rendered the list for us because a new item would have been added.

You can look up how to use this component here:
https://github.com/green-arrow/react-firestore#firestorecollection

Reply:
Cool, this is super helpful! That's exactly what we were trying to figure out how to do.